spec/lib/metasploit/credential/creation_spec.rb

```diff
@@ -178,6 +178,37 @@
         end
       end
     end
+    context 'deletion and creation' do
+      let(:private_data) { 'md5ac4bbe016b808c3c0b816981f240dcae' }
+      let(:private_data_upcase) { private_data.upcase }
+      let(:credential_data) {{
+        workspace_id: workspace.id,
+        user_id: user.id,
+        origin_type: :manual,
+        username: 'admin',
+        private_data: private_data,
+        private_type: :postgres_md5
+      }}
+      it 'creates a private cred' do
+        expect{ test_object.create_credential(credential_data) }.to change{ Metasploit::Credential::PostgresMD5.count }.by(1)
+      end
+      let(:credential_data_upcase) {{
+        workspace_id: workspace.id,
+        user_id: user.id,
+        origin_type: :manual,
+        username: 'admin',
+        private_data: private_data_upcase,
+        private_type: :postgres_md5
+      }}
+      it 'allows for the recreation of core with case insensitive private credentials set to different case' do
+        expect{ test_object.create_credential(credential_data) }.to change{ Metasploit::Credential::PostgresMD5.count }.by(1)
+        expect{ Metasploit::Credential::Core.first.destroy }.to change{ Metasploit::Credential::Core.count }.by(-1)
+        expect( Metasploit::Credential::PostgresMD5.count ).to eq(1)
+        expect( Metasploit::Credential::Core.count ).to eq(0)
+        expect{ test_object.create_credential(credential_data_upcase) }.to change{ Metasploit::Credential::Core.count }.by(1)
+        expect( Metasploit::Credential::PostgresMD5.count ).to eq(1)
+      end
```

Comment:
This matches with current expectations, should we expand `Core.destroy` to remove `public` and `private` objects that no longer have a core attached to them?

The scenario I am thinking of would be:
* add a cred with `user1` and `private1`
* add a second cred with `user2` and `private1`
* expect 2 cores,  2 publics, 1 private in db
* remove core for `user1`
* remove core for `user2`
* expect all cores, publics, private have been removed from db

And:
* add a cred with `user1` and `private1`
* add a second cred with `user2` and `private1`
* add a third cred with `user3` and `private1`
* expect 3 cores,  3 publics, 1 private in db
* remove core for `user1`
* remove core for `user2`
* expect 1 core, 1 public, 1 private are left in the db

Given the current patterns in place I can see scenarios where a core has both a `PasswordHash` and `Password` added by a cracking run. In this case when the core is deleted we would still have both privates values in the database still with no way know that the ``PasswordHash` and `Password` relate.

This could be deferred for later instead of part of this PR, just thought it is worth considering now.

Reply:
After some investigation, decision is to accept the adjustment here as is and opt to address `orphaned` data in a separate PR.